thriftbp/prometheus_test.go

```diff
@@ -0,0 +1,97 @@
+package thriftbp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate"
+)
+
+func TestPrometheusMetricsMiddleware(t *testing.T) {
+	testCases := []struct {
+		name          string
+		wantErr       thrift.TException
+		wantOK        bool
+		baseplateCode string
+	}{
+		{"bp error", thrift.NewTProtocolExceptionWithType(thrift.PROTOCOL_ERROR, WrapBaseplateError(errors.New("test"))), false, ""},
+		{"application error", thrift.NewTApplicationException(thrift.UNKNOWN_METHOD, "unknown err msg"), false, ""},
+		{"compile error", baseplate.NewError(), false, "0"},
+		{"success", nil, true, ""},
```

Comment:
also style nit: use [one-per-line style](https://github.com/reddit/baseplate.go/blob/master/Style.md#one-per-line-style) and don't use unkeyed struct constructors.

Reply:
i would prefer to leave as it if thats ok.